### PR TITLE
Tracking draft PR to remove deprecated functions

### PR DESCRIPTION
**Pull request by Ben-PH**
Added org file into this draft PR with notes, TODO's 

## Checklist 
TODO: edit this top-level comment to include things that need to be removed

- [x] CRITICAL: restore [clippys](https://github.com/seed-rs/seed/pull/453#discussion_r423548814)
- [x] `builder.rs` implementations cleared
- [x] `builder` module taken out of public API
- [x] `App::new()` in tests have init argument not Init code slated for deletion.
- [x] internal code `builder` dependencies removed
- [x] internal tests `builder` dependencies removed

- [x] `builder/after_mount.rs` implementations cleared
- [x] `builder/before_mount.rs` implementations cleared
- [x] `builder/init.rs` implementations cleared

tests pass running firefox. Chrome and safari unavailable.

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -566,25 +519,16 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
...
-
     /// App initialization: Collect its fundamental components, setup, and perform
     /// an initial render.
+    /// TODO: the `builder` module has been removed from the public interface, but
+    /// weaves a complex web of dependencies. This function is also a dependency
+    /// to [`App::start`] and other functions.
     #[deprecated(
         since = "0.4.2",
-        note = "Please use `AppBuilder.build_and_start` instead"
+        note = "Legacy note: Please use `AppBuilder.build_and_start` instead\nAs part of the 0.7.0 cleanup, this has been made private. `App::start` depends on this function."
```
Comment (MartinKavik) [inline src/app.rs:529]: I think that `run` will be also removed - don't kill time with writing nice docs for it :)

File: src/app/builder/before_mount.rs [hunk 2]
```diff
@@ -42,13 +42,13 @@ impl MountPoint for web_sys::HtmlElement {
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum MountType {
     /// Take control of previously existing elements in the mount. This does not make guarantees of
-    /// elements added after the [`App`] has been mounted.
+    /// elements added after the [`crate::app::App`] has been mounted.
```
Comment (MartinKavik) [inline src/app/builder/before_mount.rs:45]: I want to revisit all docs once a newer better way of writing links is available on stable Rust - don't spend too much time with improving documentation links.

File: src/app.rs [hunk 3]
```diff
@@ -21,7 +18,7 @@ use types::{RoutesFn, SinkFn, UpdateFn, ViewFn, WindowEventsFn};
 use wasm_bindgen::closure::Closure;
 use web_sys::Element;
 
-pub mod builder;
+pub(crate) mod builder;
```
Comment (MartinKavik) [inline src/app.rs:21]: Don't play with it, kill it! :D

File: src/lib.rs [hunk 4]
```diff
@@ -51,7 +51,12 @@
 #![allow(
     clippy::use_self,
     clippy::single_match_else,
-    clippy::must_use_candidate
+    clippy::must_use_candidate,
+
+    // temporary clearing some clippys for dev
```
Comment (MartinKavik) [inline src/lib.rs:56]: Add a big TODO flag here please, there is a big chance we'll overlook it before merging.

File: src/app.rs (hunk 3 above)
Comment (Ben-PH) [reply src/app.rs:21]: That's the plan. There are some modules underneath `builder` that breaks other code. I don't want to pull out the dependencies out from under them. I'll cull the the implementations of `builder/*.rs` line-by-line, and at the end, no builder.

File: src/lib.rs (hunk 4 above)
Comment (Ben-PH) [reply src/lib.rs:56]: I've added a check-box linking to this comment. Made the changes already, and will go into the next commit.

File: src/browser/service/storage.rs [hunk 5]
```diff
@@ -34,20 +34,3 @@ where
         .set_item(name, &serialized)
         .expect("save into `LocalStorage`");
 }
-
-/// Load a store, to a deserializable data structure.
-#[deprecated(
-    since = "0.7.0",
-    note = "Use `LocalStorage::get` or `SessionStorage::get`."
-)]
-pub fn load_data<T>(storage: &Storage, name: &str) -> Option<T>
```
Comment (MartinKavik) [inline src/browser/service/storage.rs:43]: Probably we don't have to delete this function now. I think it's out of scope of this PR and in the future the whole `service/storage` will be deleted.

Comment (Ben-PH) [reply src/browser/service/storage.rs:43]: I'll bring it back in in my next commit.

**Comment by MartinKavik** [comment]
Closing in favor of https://github.com/seed-rs/seed/pull/478

**Pull request closed without merging**